### ISSUE TITLE
Invalid DNS name in kustomization manifest

### ISSUE THREAD

**Issue by cdavisafc** [weaveworks/weave-gitops#325]
**Describe the bug**

When running a `wego app add`, by default, the kustomization object that WeGO creates for the generated GitOps automation uses the name of the kubeconfig current context. If that name is not a valid dns name this will fail.

**Environment**
 - wego: [e.g. v0.0.5]
 - kubernetes:  [e.g. 1.21]
    - [x ] KinD - _version_]
**Affects versions**

**To Reproduce**
1. Create EKS cluster with EKSCTL. By default this will create a context in your ~/.kube/config that looks something like:\
```- context:
    cluster: cornelia-prod.us-east-2.eksctl.io
    user: cornelia@weave.works@cornelia-prod.us-east-2.eksctl.io
  name: cornelia@weave.works@cornelia-prod.us-east-2.eksctl.io
```
2. Now do a `wego app add`
3. This results in an output something like:
```wego app add --path=webapp --branch=prod .
Updating parameters from environment... using URL: 'ssh://git@github.com/cdavisafc/podinfo-config.git' of origin from git config...

done

Checking cluster status... FluxInstalled

✚ deploy key: ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC1sFgq3BMDvUrQ9oJJx+faaUkIvjHw+TlI1uLGHAglxMkpajaudWWjNvVSCZnR3NOTZ8ZNerK2L78mlSJeuYNZv/IhDS4x9rsAzRISQUh6MAJkFk/sh4i77vMxfmyZAXjUq7xzXSoC3C+tsQvAbTAnAPW3OAfi0QzvrkQzhzzpKaEgRRQ5p2t12icNum+atWwfdF6+oq0J+gSUCN+bUUYi44AFt9r8yzvGbxV0+MqIplSMBq0jVU93j+8rLrrwdIfvataUY2bFSuBe+vUOp3Vo1svuL1VV3gaDianLemu3ZPSpWgoFIzv4OjDx8URWw5G8RL74QvpVV1HA0Yn2Cf+GtZAeME3+TGCe8N3O2ZTM3Bfuqe5pJD///nvvNGQGVhkFivU1d0PG5SkDjaa7JYwQa/PzN7zAV2R08CXPZa+F3Epc7qbb2mN+QK625hedLW2jJpqGt67+AxYuzTEjDWWIUuzFbQjZiTt1/dfFodsk07/K/tdc7DywpMbjqTXL1zsW+Hk3ZUkdAt3Hg8XYwyPFzLaJqK5c5zBQi9tmDxFdevTIweJ/oREI4UAF+nfC/pYqYfeGgXIF4H6U9bP2runl60vDTUgR3rqcsiuG9CkmxiW11WhsGlGGjYgvdt85V3k5cGaA6UDHLWbwN72Nl0/OGmWuwoKTvRFmmvM4joI5fw==

► secret 'podinfo-config' created in 'wego-system' namespace
---
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: GitRepository
metadata:
  name: podinfo-config
  namespace: wego-system
spec:
  interval: 30s
  ref:
    branch: prod
  secretRef:
    name: podinfo-config
  url: ssh://git@github.com/cdavisafc/podinfo-config.git

---
apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
kind: Kustomization
metadata:
  name: cornelia@weave.works@cornelia-wego-prod.us-west-2.eksctl.io-podinfo-config
  namespace: wego-system
spec:
  interval: 1m0s
  path: ./wego/targets/cornelia@weave.works@cornelia-wego-prod.us-west-2.eksctl.io
  prune: true
  sourceRef:
    kind: GitRepository
    name: podinfo-config
  validation: client

---
apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
kind: Kustomization
metadata:
  name: podinfo-config-webapp
  namespace: wego-system
spec:
  interval: 1m0s
  path: ./wego/apps/podinfo-config-webapp
  prune: true
  sourceRef:
    kind: GitRepository
    name: podinfo-config
  validation: client

could not apply manifest: ---
apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
kind: Kustomization
metadata:
  name: cornelia@weave.works@cornelia-wego-prod.us-west-2.eksctl.io-podinfo-config
  namespace: wego-system
spec:
  interval: 1m0s
  path: .wego/targets/cornelia@weave.works@cornelia-wego-prod.us-west-2.eksctl.io
  prune: true
  sourceRef:
    kind: GitRepository
    name: podinfo-config
  validation: client

: exit status 1
```

**Expected behavior**
WeGO should convert to a valid DNS name. Changing the context in the `kubeconfig` to something like the following works around the problem.

```
- context:
    cluster: cornelia-prod.us-east-2.eksctl.io
    user: cornelia@weave.works@cornelia-prod.us-east-2.eksctl.io
  name: cornelia-prod
```

**Comment by markramm** [comment]
`Warn` folks when names are not valid DNS entities.

**Comment by jpellizzari** [comment]
Was this fixed in https://github.com/weaveworks/weave-gitops/pull/508 ?

**Comment by jpellizzari** [comment]
Appears to be fixed in #508
